singularity-eos/base/fast-math/logs.hpp

```diff
@@ -67,11 +67,53 @@
  * we can go into and out of "grid space" reliably and quickly, we're
  * happy, and this function is an EXACT map into and out of that
  * space.
+ *
+ * See ArXiv:2206.08957
  */
 
 namespace singularity {
 namespace FastMath {
 
+/*
+// Integer Aliased versions. Likely not as portable
+template<typename T>
+KOKKOS_FORCEINLINE_FUNCTION
+auto sgn(const T &x) {
+  return (T(0) < x) - (x < T(0));
+}
+
+KOKKOS_FORCEINLINE_FUNCTION
+auto as_int(double f) {
+  return *reinterpret_cast<long long int*>(&f);
+}
+KOKKOS_FORCEINLINE_FUNCTION
+auto as_double(long long int i) {
+  return *reinterpret_cast<double*>(&i);
+}
```

Comment:
(this isn't very significant) I was curious about this, took a look: https://godbolt.org/z/GWGdGGvKj

If there's something I'm missing I would be interested.